node/v2/handler.js

```diff
@@ -528,9 +508,17 @@ TChannelV2Handler.prototype._handleCallFrame = function _handleCallFrame(r, fram
     if (err) {
         // TODO wrap context
         r.errorEvent.emit(r, err);
+        delete streamingColl[r.id];
         return false;
     }
 
+    if (isLast) {
+        delete streamingColl[r.id];
+    } else {
+        assert(!streamingColl[r.id] || streamingColl[r.id] === r);
```

Comment:
I presume you’ve carefully thought out and eliminated the possibility that a clever (or cosmically altered) sequence of frames could induce this assertion.